hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/TestUpdateSchemaEvolution.java

```diff
@@ -68,77 +71,232 @@ public void tearDown() throws IOException {
+          + "\"time\":\"2016-01-31T03:16:41.415Z\",\"number\":12,\"added_field\":1}";
+      List<HoodieRecord> updateRecords = new ArrayList<>();
+      RawTripTestPayload rowChange = new RawTripTestPayload(recordStr);
+      HoodieRecord record =
+          new HoodieRecord(new HoodieKey(rowChange.getRowKey(), rowChange.getPartitionPath()), rowChange);
+      record.unseal();
+      record.setCurrentLocation(new HoodieRecordLocation("101", fileId));
+      record.seal();
+      updateRecords.add(record);
+      assertThrows(HoodieUpsertException.class, () -> {
+        HoodieMergeHandle mergeHandle = new HoodieMergeHandle(config, "101", table,
+            updateRecords.iterator(), record.getPartitionPath(), fileId, supplier);
+        Configuration conf = new Configuration();
+        AvroReadSupport.setAvroReadSchema(conf, mergeHandle.getWriterSchemaWithMetafields());
+        List<GenericRecord> oldRecords = ParquetUtils.readAvroRecords(conf,
+            new Path(config.getBasePath() + "/" + insertResult.getStat().getPath()));
+        for (GenericRecord rec : oldRecords) {
+          mergeHandle.write(rec);
+        }
+        mergeHandle.close();
+      }, "UpdateFunction could not read records written with exampleSchema.txt using the "
+          + "exampleEvolvedSchemaColumnRequire.txt ,because oldrecords do not have required column added_field");
+      return 1;
+    }).collect();
+  }
+
+  @Test
+  public void testSchemaEvolutionOnUpdateMisMatchWithChangeColumnType() throws Exception {
+    final WriteStatus insertResult = prepareFirstRecordCommit(generateOneRecordForExampleSchema());
+    String fileId = insertResult.getFileId();
+
+    // Now try an update with an evolved schema
+    // Evolved schema does not have guarantee on preserving the original field ordering
+    final HoodieWriteConfig config = makeHoodieClientConfig("/exampleEvolvedSchemaColumnType.txt");
+    final HoodieSparkTable table = HoodieSparkTable.create(config, context);
+    jsc.parallelize(Arrays.asList(1)).map(x -> {
+      // New content with values for the newly added field
+      String recordStr = "{\"_row_key\":\"8eb5b87a-1feh-4edd-87b4-6ec96dc405a0\","
+          + "\"time\":\"2016-01-31T03:16:41.415Z\",\"number\":\"12\"}";
+      List<HoodieRecord> updateRecords = new ArrayList<>();
+      RawTripTestPayload rowChange = new RawTripTestPayload(recordStr);
+      HoodieRecord record =
+          new HoodieRecord(new HoodieKey(rowChange.getRowKey(), rowChange.getPartitionPath()), rowChange);
+      record.unseal();
+      record.setCurrentLocation(new HoodieRecordLocation("101", fileId));
+      record.seal();
+      updateRecords.add(record);
+      HoodieMergeHandle mergeHandle = new HoodieMergeHandle(config, "101", table,
+          updateRecords.iterator(), record.getPartitionPath(), fileId, supplier);
+      Configuration conf = new Configuration();
+      AvroReadSupport.setAvroReadSchema(conf, mergeHandle.getWriterSchemaWithMetafields());
```

Comment:
instead of creating a new Configuration, would it be better with `table.getHadoopConf()`?